### ISSUE TITLE
Agumentation /validation -files 

### ISSUE THREAD

**Issue by bmiftah** [xiaojinmao-code/DAG-V-Net#2]
Thank you for sharing this implementation.  I just want to ask where are the two files for validation.py  and the files for augtest . I don't know if it is only me but I don't see them in the list here. I read from the Readme file that part of the code is adopted from  Attention Gate model, are these files taken from that part too ?  I appreciate your reply in advance.

**Comment by xiaojinmao-code** [comment]
Hello, I did forget to upload some files. I will sort them out and upload them as soon as possible.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "xiaojinmao-code/DAG-V-Net"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月15日(星期二) 下午2:31
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[xiaojinmao-code/DAG-V-Net] Agumentation /validation -files  (Issue #2)





 
Thank you for sharing this implementation.  I just want to ask where are the two files for validation.py  and the files for augmentation . I don't know if it is only me but I don't see them in the list here. I read from the Readme file that part of the code is adopted from  Attention Gate model, are these files taken from that part too ?  I appreciate your reply in advance.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by bmiftah** [comment]
That will be great , I appreciate that very much and thank you once again.  Looking forward to see then.

**Comment by bmiftah** [comment]
@xiaojinmao-code  I come back again to check if you uploaded the remainng files ... I hope you will do it at your earliest conveinience. Thank you , stay safe out there

**Comment by xiaojinmao-code** [comment]
Yes, I have updated the code of the data enhancement part, and some are also sorting out. Because the project has been shelved for a long time, it will take some time.



---Original---
From: ***@***.***&gt;
Date: Mon, Mar 21, 2022 17:48 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [xiaojinmao-code/DAG-V-Net] Agumentation /validation -files  (Issue #2)




 
@xiaojinmao-code  I come back again to check if you uploaded the remainng files ... I hope you will do it at your earliest conveinience. Thank you , stay safe out there
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by bmiftah** [comment]
Thank you , Yes , I can guess it has been a while since it was first uploaded. I apprecaite for taking time to update part of the project.  I noticed it was ranked in the grand challange ... You did the perimeter measurement in your project - I assume this has to be part of  the submission. I am more interested in how you did the ellipse and get the related circumference. In fact , your model is undoubtly novel and its segementation result has to be pretty good one.  Looking forward whatever update you may make , I find it to be an important reference to study it,  Thank you again .

**Comment by xiaojinmao-code** [comment]
Thank you for your question, I understand that your question is about post-processing (generating those ellipse parameters needed for the race)?Because the project is on my other computer, it takes time to find it.  There are many ways，such as solution is: post-processing (corrosion, expansion) + edge extraction + minimum enclosing rectangle + minimum enclosing ellipse, mostly using the OpencV package. I will organize and upload the code later, but I have a simple implementation for reference: https://blog.csdn.net/qq_39071739/article/details/103026441?spm=1001.2014.3001.5501    I have uploaded the data of the submitted file for your reference .

**Comment by bmiftah** [comment]
Thank your reply once again. In fact, I was trying to use such code to automatically detect Ultrasound fetal head and make automatic measurement for the detected fetal head image.  In other words, I want to **detect , draw ellipse around the generated mask and measure the circumference of that ellipse - masked head and its circumference , BPD ** - that is it. I saw your project does this and I was thinking to adopt it with proper credit/quotation/ given to your work.  I have sample data to test on your model after trainng it.  I wasn't aiming to attend the race in that challange although it is an awesome idea to give it a try. 
I will refer to the link you shared for reference and I really appreciate your effort to upload the submitted data , and the files you wil be uploading at your conveinince. Thank you so much.

**Comment by xiaojinmao-code** [comment]
Ok, I will upload the code of measurement and post-processing, but it may take a week, because I can't go back to school recently.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "xiaojinmao-code/DAG-V-Net"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月24日(星期四) 下午2:45
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [xiaojinmao-code/DAG-V-Net] Agumentation /validation -files  (Issue #2)





 
Thank your reply once again. In fact, I was trying to use such code to automatically detect Ultrasound fetal head and make automatic measurement for the detected fetal head image.  In other words, I want to **detect , draw ellipse around the generated mask and measure the circumference of that ellipse - masked head and its circumference , BPD ** - that is it. I saw your project does this and I was thinking to adopt it with proper credit/quotation/ given to your work.  I have sample data to test on your model after trainng it.  I wasn't aiming to attend the race in that challange although it is an awesome idea to give it a try.
 I will refer to the link you shared for reference and I really appreciate your effort to upload the submitted data , and the files you wil be uploading at your conveinince. Thank you so much.

**Comment by bmiftah** [comment]
Okay. Thank you so much for your kind and swift reply. Perfectly understood!  In fact, your main.py file is the one important file that is currently missing . I guess the model training and other calls start from there. If you can also get this file and upload it , it wil be awesome. I will try to figure out the data path and other parts (I assume you may not keep pre-trained weights).  I can perfectly understand how difficult it is to assemble your files after such long time. But , also the best place to keep your valuable project will be here in Github so that people like me who is very much interested in your project will use it.  Many Many thanks .  Stay safe

**Comment by bmiftah** [comment]
@xiaojinmao-code - Just quick follow up on your last message, the **_main.py_** file will be great to have as it is mentioned in your README file training start from the main.py file ... I want to run and test your model as well ( I don't know if you have pre-trained model already , do you ?) . Thank you again and stay safe out there!

**Comment by bmiftah** [comment]
Hello Xiao Jin Mao ... sorry , I just saw you already uploaded the main.py file. I post the above question before checking the repository. I am sorry for that and thank you so much for being very responsive and cooperative.  I will try training your model now. I will also check out the post-processing code you uploaded. hopefully my attemp will go smooth. I will try to compare with the demo posted - I am assuming you were unable to get pre-trained model . Thanks and stay safe!
